api/v1alpha1/stardogrole_types.go

```diff
@@ -25,17 +25,32 @@ import (
 
 // StardogRoleSpec defines the desired state of StardogRole
 type StardogRoleSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	// RoleName describes (overrides) the name of a role that will be maintained in a Stardog instance.
+	// Defaults to .metadata.name.
+	RoleName string `json:"roleName,omitempty"`
+	// StardogInstanceRef references the StardogInstance object in which the role is maintained.
+	StardogInstanceRef string `json:"stardogInstanceRef,omitempty"`
+	// Permissions lists the permissions assigned to a role
+	Permissions []StardogPermissionSpec `json:"permissions,omitempty"`
+}
 
-	// Foo is an example field of StardogRole. Edit StardogRole_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+// StardogPermissionSpec defines a Stardog permission assigned to a Role
+type StardogPermissionSpec struct {
+	// Action describes the action a specific permission is assigned to
+	// +kubebuilder:validation:Enum=All;Create;Delete;Read;Write;Grant;Revoke;Execute
+	Action string `json:"action,omitempty"`
+	// ResourceType describes the type of resource a specific permission is assigned to
+	// +kubebuilder:validation:Enum=DB;USER;ROLE;ADMIN;METADATA;NAMEDGRAPH;VIRTUALGRAPH;ICVCONTRAINTS
```

Comment:
I suggest to use Capital case like in Action except for `DB` (or better write it out, "Database"). Also, you're sure there's no typo in `ICVCONTRAINTS` (missing S) ?
```suggestion
	// +kubebuilder:validation:Enum=Database;User;Role;Admin;Metadata;NamedGraph;VirtualGraph;IcvConstraints
```

Reply:
I would leave the values as is, like in Stardog. Will correct the name though `ICVCONTRAINTS`

Reply:
What is your reference point? In the API they are lower case e.g: https://stardog.docs.apiary.io/#reference/admin/roles/put

Reply:
Querying the Database using the Stardog Studio. I will capitalize the other enum.

Reply:
Just updated again everything, is it ok for you if I merge?